<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Dustin Sokolowski" />

<meta name="date" content="2023-02-17" />

<title>single cell Mapper (scMappR)</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">single cell Mapper (scMappR)</h1>
<h4 class="author">Dustin Sokolowski</h4>
<h4 class="date">2023-02-17</h4>


<div id="TOC">
<ul>
<li><a href="#install-and-load-scmappr" id="toc-install-and-load-scmappr">Install and load scMappR</a></li>
<li><a href="#downloading-files." id="toc-downloading-files.">Downloading Files.</a></li>
<li><a href="#saving-results." id="toc-saving-results.">Saving
results.</a></li>
<li><a href="#introduction-of-the-primary-functions-highlighted-in-the-vignette." id="toc-introduction-of-the-primary-functions-highlighted-in-the-vignette.">Introduction
of the primary functions highlighted in the vignette.</a></li>
<li><a href="#cell-weighted-fold-changes-cwfold-changes-generation" id="toc-cell-weighted-fold-changes-cwfold-changes-generation">cell-weighted
Fold Changes (cwFold-changes) Generation</a>
<ul>
<li><a href="#saved-outputs" id="toc-saved-outputs">Saved
outputs</a></li>
<li><a href="#additional-pathway-enrichments-of-cwfold-changes" id="toc-additional-pathway-enrichments-of-cwfold-changes">Additional
pathway enrichments of cwFold-changes</a></li>
</ul></li>
<li><a href="#testing-the-cell-type-specificity-of-cwfold-changes-at-the-level-of-the-gene-and-cell-type." id="toc-testing-the-cell-type-specificity-of-cwfold-changes-at-the-level-of-the-gene-and-cell-type.">Testing
the cell-type specificity of cwFold-changes at the level of the gene and
cell-type.</a></li>
<li><a href="#cell-type-markers-in-a-list-of-genes." id="toc-cell-type-markers-in-a-list-of-genes.">Cell-type markers in a
list of genes.</a>
<ul>
<li><a href="#no-custom-signature-matrix" id="toc-no-custom-signature-matrix">No custom signature matrix:</a></li>
<li><a href="#custom-signature-matrix" id="toc-custom-signature-matrix">Custom signature matrix</a></li>
<li><a href="#saved-outputs-2" id="toc-saved-outputs-2">Saved
Outputs</a></li>
</ul></li>
<li><a href="#tissue-by-cell-type-enrichment." id="toc-tissue-by-cell-type-enrichment.">Tissue by cell-type
enrichment.</a>
<ul>
<li><a href="#three-types-of-outputs." id="toc-three-types-of-outputs.">Three types of outputs.</a></li>
</ul></li>
<li><a href="#processing-scrna-seq-count-data-into-a-signature-matrix." id="toc-processing-scrna-seq-count-data-into-a-signature-matrix.">Processing
scRNA-seq count data into a signature matrix.</a>
<ul>
<li><a href="#processing-scrna-seq-data-with-multiple-scrna-seq-runs." id="toc-processing-scrna-seq-data-with-multiple-scrna-seq-runs.">Processing
scRNA-seq data with multiple scRNA-seq runs.</a></li>
<li><a href="#processing-scrna-seq-count-data-from-a-different-species." id="toc-processing-scrna-seq-count-data-from-a-different-species.">Processing
scRNA-seq count data from a different species.</a></li>
<li><a href="#processing-scrna-seq-count-data-when-cell-types-are-already-named." id="toc-processing-scrna-seq-count-data-when-cell-types-are-already-named.">Processing
scRNA-seq count data when cell-types are already named.</a></li>
<li><a href="#signature-matrix-from-seurat-object-with-named-cell-type" id="toc-signature-matrix-from-seurat-object-with-named-cell-type">Signature
matrix from Seurat object with named cell-type</a></li>
</ul></li>
<li><a href="#manually-making-graphics." id="toc-manually-making-graphics.">Manually making graphics.</a>
<ul>
<li><a href="#barplots" id="toc-barplots">Barplots</a></li>
<li><a href="#heatmaps" id="toc-heatmaps">Heatmaps</a></li>
</ul></li>
</ul>
</div>

<div id="install-and-load-scmappr" class="section level2">
<h2>Install and load scMappR</h2>
<p>scMappR relies on the following dependencies which should be
downloaded/updated with scMappR automatically. Please ensure that these
packages are not open when installing scMappR.</p>
<ul>
<li>ggplot2 - CRAN</li>
<li>pheatmap - CRAN</li>
<li>graphics - CRAN</li>
<li>Seurat - CRAN</li>
<li>GSVA - Bioconductor</li>
<li>stats - CRAN</li>
<li>utils - CRAN</li>
<li>downloader - CRAN</li>
<li>pcaMethods - Bioconductor</li>
<li>grDevices - CRAN</li>
<li>gProfileR - CRAN</li>
<li>limSolve - CRAN</li>
<li>ADAPTS - CRAN</li>
<li>reshape - CRAN</li>
</ul>
<p>Install GSVA and pcaMethods from bioconductor first, as
<code>devtools::install_githb()</code> will automatically install
CRAN.</p>
<ol style="list-style-type: decimal">
<li>Github (Development Version)</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;devtools&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;pcaMethods&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GSVA&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;wilsonlabgroup/scMappR&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>CRAN (Stable Release)</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;devtools&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;pcaMethods&quot;</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GSVA&quot;</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;scMappR&quot;</span>)</span></code></pre></div>
</div>
<div id="downloading-files." class="section level2">
<h2>Downloading Files.</h2>
<p>Before using scMappR, we strongly recommend that you download data
files from “<a href="https://github.com/wilsonlabgroup/scMappR_Data" class="uri">https://github.com/wilsonlabgroup/scMappR_Data</a>”.Tthe
<code>rda_path</code> parameter will allow users to set the directory
where these data are stored. If the correct files are not stored in this
directory, then scMappR will download these as temporary files from “<a href="https://github.com/wilsonlabgroup/scMappR_Data" class="uri">https://github.com/wilsonlabgroup/scMappR_Data</a>”, use
them in the function, and not save them. This parameter is set to “” as
a default and thus data are downloaded as a default.</p>
<p>Additionally, the <code>get_signature_matrices()</code> function
allows the user to get the list of all stored signature matrices at any
given time. Input “all”, “pVal”, or “OR” to get the matrix of p-values,
odds-ratios, or both. This function outputs the signature matrices, the
top 30 cell-type markers for each cell-type, and the predicted cluster
label from the Fisher’s exact test and GSVA.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>signatures <span class="ot">&lt;-</span> <span class="fu">get_signature_matrices</span>(<span class="at">type =</span> <span class="st">&quot;all&quot;</span>) <span class="co">#return a list of cell-type labels, p-values, and odds-ratios.</span></span></code></pre></div>
</div>
<div id="saving-results." class="section level2">
<h2>Saving results.</h2>
<p>Many of the functions in scMappR print files or generate directories.
To return the full output of scMappR, please change the ‘toSave’
parameters from <code>FALSE</code> to <code>TRUE</code> in any of the
functions being used. Otherwise, the functions in scMappR will only
return a small portion of what scMappR has to offer. Due to CRAN
packages not allowing for their packages to print files/make
directories, toSave is set to <code>FALSE</code> as default.
Furthermore, a path directory must be set. For example
<code>path=&quot;./&quot;</code> will print files and directories to the working
directory. These measures are to prevent any file overwriting. This
vignette sets <code>path</code> to <code>tempdir()</code>. When using
scMappR, change <code>tempdir()</code> to the directories where files
should be saved.</p>
</div>
<div id="introduction-of-the-primary-functions-highlighted-in-the-vignette." class="section level2">
<h2>Introduction of the primary functions highlighted in the
vignette.</h2>
<div id="transforming-summary-statistics-of-differentially-expressed-genes-by-cell-type-specific-information" class="section level4">
<h4>Transforming summary statistics of differentially expressed genes by
cell-type specific information</h4>
<ul>
<li><code>scMappR_and_pathway_analysis()</code>: This function generates
cell-weighted fold changes (cellWeighted_Foldchanges), visualizes them
in a heatmap, and completes pathway enrichment of
cellWeighted_Foldchanges and bulk gene list.</li>
<li><code>two_method_pathway_enrichment()</code>: This function
completes additional pathway enrichment tests from cwFold-changes
calculated in <code>scMappR_and_pathway_analysis</code>. Specifically,
it re-ranks cwFoldChanges based on their absolute ct specificity scores
(per-celltype) as well as their rank increase in cell-type specificity
before completing an ordered pathway analysis. In the second method,
only genes with a rank increase in cell-type specificity were
included.</li>
<li><code>cwFoldChange_evaluate()</code>. At the level of the gene,
cwFold-changes are scored so that they sum to 1. For each gene,
cell-types whose cwFold-change are greater than the cell-type proportion
while accounting for an abnormally high proportion of the fold change (+
3 median absolute deviations from median cell-type specificity) are
considered cell-type specific. At the level of the cell-type, bulk DEGs
and cwFold-changes for each cell-type are correlated. The difference in
the rank of DEG is also measured.</li>
</ul>
</div>
<div id="enriching-cell-type-markers-in-a-list-of-genes." class="section level4">
<h4>Enriching cell-type markers in a list of genes.</h4>
<ul>
<li><code>tissue_scMappR_custom()</code>: This function visualizes
signature matrix, clusters subsetted genes, completes enrichment of
individual cell-types and co-enrichment.</li>
<li><code>tissue_scMappR_internal()</code>: This function loops through
every signature matrix in a particular tissue and generates heatmaps,
cell-type preferences, and co-enrichment.</li>
<li><code>tissue_by_celltype_enrichment()</code>: This function
completes a fishers exact test of an input list of genes against one of
the two curated tissue by cell-type marker datasets from scMappR.</li>
</ul>
</div>
<div id="processing-scrna-seq-count-data-into-a-gene-list." class="section level4">
<h4>Processing scRNA-seq count data into a gene list.</h4>
<ul>
<li><code>process_dgTMatrix_lists()</code>: This function takes a list
of count matrices, processes them, calls cell-types, and generates
signature matrices.</li>
</ul>
</div>
</div>
<div id="cell-weighted-fold-changes-cwfold-changes-generation" class="section level2">
<h2>cell-weighted Fold Changes (cwFold-changes) Generation</h2>
<p>The <code>scMappR_and_pathway_analysis()</code> function generates
cellWeighted_Foldchanges based on an inputted signature matrix,
normalized RNA-seq count matrix (e.g. TPM, RPKM, CPM), and list of
differentially-expressed genes (DEGs) before creating cell-weighted Fold
Changes (cwFold-changes). cwFold-changes are row-normalized and
visualized in a heatmap. Cell-type markers within calculated
cwFold-changes are also visualized both as cell-type markers on their
own (i.e. just their cell-type specificity), and as cwFold-changes.
Then, for each cell-type, this function filters genes expressed in each
cell-type (cwFold-change &gt; 1e-5), reorders the genes by
cwFold-change, then completes an ordered pathway analysis with
g:ProfileR or gprofiler2 package (user chooses). All genes detected in
the bulk RNA-seq are used as the background.   The example below has
<code>toSave = FALSE</code>, <code>up_and_downregulated = FALSE</code>
and <code>internet = FALSE</code>. When running scMappR yourself, it is
strongly recommended to set all to <code>TRUE</code>.
<code>toSave = TRUE</code> allows for the printing of folders, images,
and files onto a desktop/cluster.
<code>up_and_downregulated = TRUE</code> repeats pathway analysis for up
and down-regulated genes separately. <code>internet = TRUE</code> allows
for the completion of pathway analysis altogether. Tissue-types are
available in <code>data(scMappR_tissues)</code>, and the signature
matrices themselves from
<code>get_signature_matrices(type = &quot;OR&quot;)</code>.</p>
<div id="example" class="section level4">
<h4>Example</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(PBMC_scMappR) <span class="co"># load data example of PBMC bulk- and cell-sorted RNA-seq data</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>bulk_DE_cors <span class="ot">&lt;-</span> PBMC_example<span class="sc">$</span>bulk_DE_cors <span class="co"># 59 sex-specific DEGs in bulk PBMC (up-regulated = female-biased)</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>bulk_normalized <span class="ot">&lt;-</span> PBMC_example<span class="sc">$</span>bulk_normalized <span class="co"># log CPM normalized bulk RNA-seq data</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>odds_ratio_in <span class="ot">&lt;-</span> PBMC_example<span class="sc">$</span>odds_ratio_in <span class="co"># signature matrix developed from cell-sorted RNA-seq</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>case_grep <span class="ot">&lt;-</span> <span class="st">&quot;_female&quot;</span> <span class="co"># flag for &#39;cases&#39; (up-regulated), index is also acceptable</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>control_grep <span class="ot">&lt;-</span> <span class="st">&quot;_male&quot;</span> <span class="co"># flag for &#39;control&#39; (down-regulated), index is also acceptable</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>max_proportion_change <span class="ot">&lt;-</span> <span class="dv">10</span> <span class="co"># maximum cell-type proportion change -- this is good for cell-types that are uncomon in population and small absolute changes may yield large relative changes</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>theSpecies <span class="ot">&lt;-</span> <span class="st">&quot;human&quot;</span> <span class="co"># these RNA-seq data have human gene symbols (and are also from human)</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co"># When running scMappR, it is strongly recommended to use scMappR_and_pathway analysis with the parameters below.</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>toOut <span class="ot">&lt;-</span> <span class="fu">scMappR_and_pathway_analysis</span>(bulk_normalized, odds_ratio_in, </span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>                                      bulk_DE_cors, <span class="at">case_grep =</span> case_grep,</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">control_grep =</span> control_grep, <span class="at">rda_path =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">max_proportion_change =</span> <span class="dv">10</span>, <span class="at">print_plots =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">plot_names =</span> <span class="st">&quot;scMappR_vignette_&quot;</span>, <span class="at">theSpecies =</span> <span class="st">&quot;human&quot;</span>, </span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">output_directory =</span> <span class="st">&quot;scMappR_vignette_&quot;</span>,</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sig_matrix_size =</span> <span class="dv">3000</span>, <span class="at">up_and_downregulated =</span> <span class="cn">TRUE</span>, </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">internet =</span> <span class="cn">TRUE</span>, <span class="at">toSave =</span> <span class="cn">TRUE</span>, <span class="at">path =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
</div>
<div id="output" class="section level4">
<h4>Output</h4>
</div>
<div id="saved-outputs" class="section level3">
<h3>Saved outputs</h3>
<p>Assuming <code>toSave = T</code> and
<code>up_and_downregulated = T</code>,
<code>scMappR_and_pathway_analysis()</code> will also generate a folder
in your current directory with a considerable amount of
data/figures.</p>
<p>Here, we will walk through the output of the above example sorting by
name and working downwards.</p>
<ul>
<li>BP_barplot: a directory with pdf’s of the top 10 pathways of DEGs
that have been re-normlaized by their cwFold-change.</li>
<li>Downregulated: a directory with pdf’s and Rdata files of pathway and
TF enrichment of downregulated DEGs re-normalized by their
cwFold-change.</li>
<li>TF_barplot: a directory with pdf’s of the top 10 TFs of DEGs that
have been re-normlaized by their cwFold-change.</li>
<li>Upregulated: a directory with pdf’s and Rdata files of pathway and
TF enrichment of upregulated DEGs re-normalized by their
cwFold-change.</li>
<li>bulk_pathway_enrichment: png and Rdata file of traditional g:ProfilR
analysis of GOBP, REAC, and KEGG databases.</li>
<li>bulk_TF_enrichment: png and Rdata file of traditional g:ProfilR
analysis of TF database.</li>
<li>scMappR_vignette_cell_proportion_changes_summary: t-tests looking at
difference in estimated cell-type proporitons between conditions.</li>
<li>scMappR_vignette_celltype_proportions: Estimated cell-type
proportions of all inputted samples.</li>
<li>scMappR_vignette_cellWeighted_Foldchanges: cwFold-changes of all
estimated DEGs.</li>
<li>scMappR_vignette_cwFC_downregulated_DEGs: heatmap of downregulated
cwFold-changes.</li>
<li>scMappR_vignette_cwFC_signature_matrix_downregulated_DEGs: heatmap
of downregulated cwFold-changes overlapping with signature matrix.</li>
<li>scMappR_vignette_DEG_signature_matrix: Heatmap of signature
matrix.</li>
<li>scMappR_vignette_leaveOneOut_gene_proportions: estimated cell-type
proportions of all samples after each DEG is removed.</li>
<li>scMappR_vignette_reordered_pathways: g:ProfileR enrichment of DEGs
after they’re re-ordered by cwFold-changes.<br />
</li>
<li>scMappR_vignette_reordered_transcription_factors: g:ProfileR
enrichment of DEGs after they’re re-ordered by cwFold-changes.<br />
</li>
<li>scMappR_vignette_signature_matrix: heatmap of signature matrix.</li>
<li>scMappR_vignette_signature_matrix_downregulated_DEGs: heatmap of
signature matrix overlapping downregulated DEGs.</li>
<li>scMappR_vignette_signature_matrix_upregulated_DEGs: heatmap of
signature matrix overlapping upregulated DEGs.</li>
<li>scMappR_vignetteabs_cwFC: heatmap of absolute valued cwFold-changes
(up- and down-regulated DEGs).</li>
<li>scMappR_vignettecwFC_signature_matrix_upregulated_DEGs: heatmap of
absolute valued cwFold-changes (up- and down-regulated DEGs) overlapping
with the signature matrix.</li>
</ul>
</div>
<div id="additional-pathway-enrichments-of-cwfold-changes" class="section level3">
<h3>Additional pathway enrichments of cwFold-changes</h3>
<p>In addition to completing pathway enrichment of DEGs re-ranked by
their cwFold-change, it may be informative to interrogate which DEGs
were the most influenced by their cell-type specificity in the context
of each cell-type.</p>
<div id="example-1" class="section level4">
<h4>Example</h4>
<p>In this example, we are extending the output from the
<code>scMappR_and_pathway_analysis</code> function above.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>twoOutFiles <span class="ot">&lt;-</span> <span class="fu">two_method_pathway_enrichment</span>(bulk_DE_cors, <span class="st">&quot;human&quot;</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="at">scMappR_vals =</span> toOut<span class="sc">$</span>cellWeighted_Foldchange, <span class="at">background_genes =</span> <span class="fu">rownames</span>(bulk_normalized), </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="at">output_directory =</span> <span class="st">&quot;newfun_test&quot;</span>,<span class="at">plot_names =</span> <span class="st">&quot;nonreranked_&quot;</span>, <span class="at">toSave =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="output-1" class="section level4">
<h4>Output</h4>
<ul>
<li>rank_increase: a list containing the rank change between DEGs and
cwFold-changes. Additionally, it contains the output of g:ProfileR with
genes re-ranked by eac cwFold-change.</li>
<li>non_rank_increase: the same output of pathway enrichment found in
<code>scMappR_and_pathway_analysis</code>. Please refer to previous
section for output.</li>
</ul>
<div id="saved-outputs-1" class="section level5">
<h5>Saved outputs</h5>
<ul>
<li>BP_barplot: Plots of the top 10 most enriched pathways by re-ranking
genes by cwFold-change using g:ProfileR.</li>
<li>TF_barplot: Plots of the top 10 most enriched TFs by re-ranking
genes by cwFold-change using g:ProfileR.</li>
<li>bp_reranked: Plots of the top 10 most enriched pathways by
re-ordering genes based on how much the gene increases in rank between
Fold-change and cwFold-change.</li>
<li>tf_reranked: Plots of the top 10 most enriched TFs by re-ordering
genes based on how much the gene increases in rank between Fold-change
and cwFold-change.</li>
<li>scatter_reranked: Scatter plots of the rank of cwFold-changes and
regulated fold-changes of DEGs for each cell-type.</li>
<li>bulk_X_enrichment.Rdata: gProfiler results of pathway/TF enrichment
of DEGs.</li>
<li>rank_increase_genes.Rdata: list showing the rank difference between
Fold-changes and cwFold-changes for each cell-type.</li>
<li>reweighted_bulk_X.Rdata: List of each cell-type containing
Pathway/TF enrichment of genes re-ordered by cwFold-change.</li>
<li>reweighted_reordered_X.Rdata: List of each cell-type containing
Pathway/TF enrichment of genes ordered by the change in rank between
fold-change and cwFold-change.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="testing-the-cell-type-specificity-of-cwfold-changes-at-the-level-of-the-gene-and-cell-type." class="section level2">
<h2>Testing the cell-type specificity of cwFold-changes at the level of
the gene and cell-type.</h2>
<p>An important aspect of scMappR is to calculate which DEGs from bulk
differential analysis may be differentially expressed due to a change in
expression of specific cell-types. We complete this task my measuring if
the cell-type specificity of each DEG contains outliers for a specific
cell-type, and if the distribution of DEGs within a cell-type (measured
through cwFold-changes) is different from the distribution of DEGs
measured in bulk.</p>
<div id="example-2" class="section level4">
<h4>Example</h4>
<p>In this example, we are extending the output from the
<code>scMappR_and_pathway_analysis</code> function above.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>evaluated <span class="ot">&lt;-</span> <span class="fu">cwFoldChange_evaluate</span>(toOut<span class="sc">$</span>cellWeighted_Foldchange, toOut<span class="sc">$</span>cellType_Proportions, bulk_DE_cors)</span></code></pre></div>
</div>
<div id="output-2" class="section level4">
<h4>Output</h4>
<ul>
<li>cwFoldChange_normalized: A gene by cell-type data.frame of
cwFold-changes. Here, the cwFold-change of each gene is divided by it’s
sum such that the cwFold-change can be represented as how much that gene
is accounting for the fold-change of the DEG.</li>
<li>gene_level_investigation: A data.frame where each row is a gene. The
first column is that gene’s normalized cwFold-change’s (see previous
point) euclidean distance from 1. The following columns test if that
gene’s normalized cwFold-change fits a normal distribution using a
Shapiro test.</li>
<li>cwFoldchange_gene_assigned: A list of cell-types. Each cell-type
contains genes where scMappR would predict that the cell-type is driving
the gene’s expression in the researcher’s bulk sample. For each gene,
outliers are measured using mean + 3sd or median + 3mad depending on if
the cwFold-change of the gene fit the normal distribution. As a second
threshold, the cwFold-change of the DEG in the outlier cell-type needed
needed to be greater than the cell-type proportion of that cell-type.
The normalized cwFold-change for that gene in the assigned cell-type is
also stored.
<ul>
<li>cwFoldchange_gene_flagged_FP: In our cell-type proportion scaling
step, we multiply the fold-change of each DEG by the reciprical of the
cell-type proporitons between case and controls. In the condition when
there are many DEGs that do not overlap with the signature matrix and
there are one or more cell-types with a very large difference in
cell-type proportion. The scaling step can overpredit and assign
non-celltype specific DEGs to a cell-type based on the change in
cell-type proportion. These DEGs have a very predictable pattern and are
stored in the <code>cwFoldchange_gene_flagged_FP</code> list. There is a
possibility that these genes are not false positives, but should be
inspected further by the researcher if they are to be included in
downstream analysis. By default we suggest that these genes are not
considered cell-type specific.</li>
</ul></li>
<li>celltype_level_investigation: data frame where the columns are
cell-type and the rows are the Pearson and Spearman correlation
coefficients (and p-values) between bulk DEGs and cwFold-changes.</li>
<li>cwFoldchange_vs_bulk_rank_change: A gene by cell-type matrix
populated with what rank the gene is differentially expressed given that
cell-type.</li>
</ul>
</div>
</div>
<div id="cell-type-markers-in-a-list-of-genes." class="section level2">
<h2>Cell-type markers in a list of genes.</h2>
<p>Given a tissue and an unranked list of genes (i.e. without a count
matrix or summary statistics), the <code>tissue_scMappR_custom()</code>
and <code>tissue_scMappR_internal()</code> functions visualizes
cell-type markers contained within the gene-list. Then, they test for
the enrichment of cell-types within that tissue. When there is no custom
signature matrix, providing a tissue present in
<code>get_signature_matrices(type = &quot;pVal&quot;)</code> will complete
cell-type specific gene visualization and enrichment for every signature
matrix in the tissue.</p>
<div id="no-custom-signature-matrix" class="section level3">
<h3>No custom signature matrix:</h3>
<div id="example-3" class="section level4">
<h4>Example</h4>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(POA_example) <span class="co"># region to preoptic area</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>Signature <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_Rank_signature <span class="co"># signature matrix </span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>rowname <span class="ot">&lt;-</span> <span class="fu">get_gene_symbol</span>(Signature) <span class="co"># get signature</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Signature) <span class="ot">&lt;-</span> rowname<span class="sc">$</span>rowname</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Signature)[<span class="dv">1</span><span class="sc">:</span><span class="dv">60</span>]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>rda_path1 <span class="ot">=</span> <span class="st">&quot;&quot;</span> <span class="co"># data directory (if it exists)</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify tissues available for tissue_scMappR_internal</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(scMappR_tissues)</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hypothalamus&quot;</span> <span class="sc">%in%</span> <span class="fu">toupper</span>(scMappR_tissues)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>internal <span class="ot">&lt;-</span> <span class="fu">tissue_scMappR_internal</span>(genes, <span class="st">&quot;mouse&quot;</span>, <span class="at">output_directory =</span> <span class="st">&quot;scMappR_Test_Internal&quot;</span>, </span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="at">tissue =</span> <span class="st">&quot;hypothalamus&quot;</span>, <span class="at">rda_path =</span> rda_path1, <span class="at">toSave =</span> <span class="cn">TRUE</span>, <span class="at">path =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
</div>
<div id="output-3" class="section level4">
<h4>Output</h4>
<p>This function returns a list “internal”. Using the example of
<code>internal[[4]]</code>, the pre-optic area of the hypothalamus, you
can see four objects.</p>
<div id="background_heatmap" class="section level5">
<h5>background_heatmap</h5>
<ul>
<li>genesIn: genes in the signature matrix is tested.<br />
</li>
<li>genesNoIn: genes not in the signature matrix (empty).<br />
</li>
<li>geneHeat: Gene x cell-type matrix containing the cell-type rank of
each gene in the signature matrix.<br />
</li>
<li>preferences: Cell-type marker genes sorted into each cell-type.</li>
</ul>
</div>
<div id="gene_list_heatmap" class="section level5">
<h5>gene_list_heatmap</h5>
<ul>
<li>genesIn: Inputted genes that are also in the signature matrix.</li>
<li>genesNoIn: Inputted genes not in the signature matrix.</li>
<li>geneHeat: Gene x cell-type matrix containing the cell-type rank for
inputted genes overlaying with genes in the signature matrix.</li>
<li>preferences: Inputted genes overlaying with signature matrix sorted
into the cell-types where they’re preferentially expressed.</li>
</ul>
</div>
<div id="single_celltype_preferences" class="section level5">
<h5>single_celltype_preferences</h5>
<ul>
<li>Output of the Fisher’s-exact test for cell-type enrichment of
inputted genes with every cell-type.</li>
</ul>
</div>
<div id="group_celltype_preference" class="section level5">
<h5>group_celltype_preference</h5>
<ul>
<li>Identification and statistical enrichment of groups of cell-types
containing the same cell-type marker genes.</li>
</ul>
</div>
</div>
<div id="saved-directory" class="section level4">
<h4>Saved directory</h4>
<p>When <code>toSave == TRUE</code>, a directory is generated with
visualization of each signature matrix, the signature matrix subsetted
by input genes, and statistical enrichment.</p>
</div>
</div>
<div id="custom-signature-matrix" class="section level3">
<h3>Custom signature matrix</h3>
<p><code>tissue_scMappR_custom()</code> assumes a custom signature
matrix. We suggest a gene x cell-type matrix populated with the
<code>-log10(Padj)*sign(FC)</code> (i.e. Rank) of the cell-type marker.
We identify the cell-type marker with the <code>FindMarker</code>
function from the Seurat package. The statistical test within that
function is based on the biological question that the user interested
in. Pre-computed signature matrices use the Wilcoxon test. If the value
filling the matrix is not rank, users will need to change the
<code>gene_cutoff</code> parameter to what they deem significant. If
<code>is_pvalue == TRUE</code>, <code>tissue_scMappR_custom()</code>
will transform the p-values into Ranks.</p>
<div id="example-4" class="section level4">
<h4>Example</h4>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Acquiring the gene list</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(POA_example)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>Signature <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_Rank_signature</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>rowname <span class="ot">&lt;-</span> <span class="fu">get_gene_symbol</span>(Signature)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Signature) <span class="ot">&lt;-</span> rowname<span class="sc">$</span>rowname</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Signature)[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>]</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#running tisue_scMappR_custom</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>internal <span class="ot">&lt;-</span> <span class="fu">tissue_scMappR_custom</span>(genes,Signature,<span class="at">output_directory =</span> <span class="st">&quot;scMappR_Test_custom&quot;</span>, <span class="at">toSave =</span> F)</span></code></pre></div>
</div>
<div id="output-4" class="section level4">
<h4>Output</h4>
<p>The output is identical to that found in
<code>tissue_scMappR_internal()</code> but with only one study, or the
example shown above.<br />
</p>
<p>If you choose to set <code>toSave = TRUE</code>. This function will
return a directory with a number of relevant files.</p>
</div>
</div>
<div id="saved-outputs-2" class="section level3">
<h3>Saved Outputs</h3>
<ul>
<li>_celltype_preferences.tsv: Enrichment of inputted genes on each
cell-type.</li>
<li>_cell_co_preferences.tsv: Identification and enrichment of
cell-types being enriched by the same genes (cell-type markers in
common).</li>
<li>_custom_background_heatmap.pdf: Heatmap of all cell-type markers in
each cell-type in the entire signature matrix.</li>
<li>_custom_background_preferences.pdf: All cell-type markers sorted
into each cell-type.</li>
<li>_custom_genelist_heatmap.pdf: Heatmap of cell-type markers
intersected with inputted gene list.</li>
<li>_custom_genelist_preferences.pdf: Cell-type markers intersected with
inputted gene list sorted into each cell-type.</li>
</ul>
<div id="saved-directory-1" class="section level4">
<h4>Saved directory</h4>
<p>When <code>toSave == TRUE</code>, a directory is generated with
visualization of each signature matrix, the signature matrix subsetted
by input genes, and statistical enrichment.</p>
</div>
</div>
</div>
<div id="tissue-by-cell-type-enrichment." class="section level2">
<h2>Tissue by cell-type enrichment.</h2>
<p>scMappR allows for gene-set enrichment of every cell-type marker,
sorted by cell-type, tissue, and study, curated while preprocessing all
of the signature matrices for the functions within scMappR. The dataset
may be downloaded from <a href="https://github.com/DustinSokolowski/scMappR_Data" class="uri">https://github.com/DustinSokolowski/scMappR_Data</a>,
processed into a gmt file using a number of packages (qusage,
activepathways etc.) and then used with traditional gene-set-enrichment
tools (GSEA, GSVA, gprofiler). Additionally, scMappR contains the
<code>tissue_by_celltype_enrichment</code> function, that enriches a
gene list against all cell-type markers using a Fisher’s-exact test.
Users can also download all cell-type markers by setting
<code>return_gmt = TRUE</code> when using the function.</p>
<div id="example-5" class="section level4">
<h4>Example</h4>
<p>Here, we will investigate the tissue x cell-type enrichment of the
top 100 genes in the Preoptic area signature matrix.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(POA_example)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>POA_generes <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_generes</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>POA_OR_signature <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_OR_signature</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>POA_Rank_signature <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_Rank_signature</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>Signature <span class="ot">&lt;-</span> POA_Rank_signature</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>rowname <span class="ot">&lt;-</span> <span class="fu">get_gene_symbol</span>(Signature)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Signature) <span class="ot">&lt;-</span> rowname<span class="sc">$</span>rowname</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Signature)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>enriched <span class="ot">&lt;-</span> <span class="fu">tissue_by_celltype_enrichment</span>(<span class="at">gene_list =</span> genes, </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="at">species =</span> <span class="st">&quot;mouse&quot;</span>,<span class="at">p_thresh =</span> <span class="fl">0.05</span>, <span class="at">isect_size =</span> <span class="dv">3</span>)</span></code></pre></div>
</div>
<div id="three-types-of-outputs." class="section level3">
<h3>Three types of outputs.</h3>
<ul>
<li>tissue x CT enrichment: Always returned when using the
<code>tissue_by_celltype_enrichment()</code> function. This function
returns gene set enrichment compatible with <code>plotBP()</code> for
all cell-type markers significantly enriched via the function.</li>
<li>gmt_file: Since the bank of signature matrices is updated monthly,
it is recommended to periodically download a new
human_tissue_celltype_scMappR.rda and mouse_tissue_celltype_scMappR.rda
from <a href="https://github.com/DustinSokolowski/scMappR_Data" class="uri">https://github.com/DustinSokolowski/scMappR_Data</a>. When
internet is available, setting “rda_path” to “” will download the most
updated pathway files directly using the downloader package. Here, if
<code>return_gmt == TRUE</code>, then this downloaded gmt will be
returned with the enrichment.</li>
<li>if <code>toSave == TRUE</code>, then the -log10(P_adj) of the top 10
tissue/cell-types that are enriched are plotted.</li>
</ul>
</div>
</div>
<div id="processing-scrna-seq-count-data-into-a-signature-matrix." class="section level2">
<h2>Processing scRNA-seq count data into a signature matrix.</h2>
<p>scMappR inputs a list of count matrices (of class list, dCGMatrix, or
matrix) and re-processes it using the standard Seurat V4 vignette (+
removal of cells with &gt; 2 standard deviations of mt contamination
than mean). Then, it finds cell-type markers and identifies potential
cell-type names using the GSVA and Fisher’s exact methods on the
CellMarker and Panglao databases. Finally, it creates a signature matrix
of odds ratios and ranks. There are options to save the Seurat object,
GSVA cell-type identities and list of cell-type markers. To identify
what naming-preferences options described here: “brain”, “epithelial”,
“endothelial”, “blood”, “connective”,“eye”, “epidermis”, “Digestive”,
“Immune”, “pancreas”, “liver”, “reproductive”, “kidney”,
“respiratory”</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sm)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>toProcess <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">example =</span> sm)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>tst1 <span class="ot">&lt;-</span> <span class="fu">process_dgTMatrix_lists</span>(toProcess, <span class="at">name =</span> <span class="st">&quot;testProcess&quot;</span>, <span class="at">species_name =</span> <span class="st">&quot;mouse&quot;</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="at">naming_preference =</span> <span class="st">&quot;eye&quot;</span>, <span class="at">rda_path =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="at">toSave =</span> <span class="cn">TRUE</span>, <span class="at">saveSCObject =</span> <span class="cn">TRUE</span>, <span class="at">path =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
<p>It is recommended to set <code>toSave == TRUE</code>, allowing for
important data objects to be saved. Here, the above function is repeated
with <code>toSave == TRUE</code> and <code>saveSCObject == TRUE</code>,
and the outputted files will be briefly discussed.</p>
<p>Here, the following objects are saved.</p>
<ul>
<li>testProcess_generes.Rdata: list of cell-type markers for every
cluster.</li>
<li>testProcess_or_heatmap.Rdata: signature matrix of odds ratios named
from Fisher’s exact test.</li>
<li>testProcess_pval_heatmap.Rdata: signature matrix of ranks named from
Fisher’s exact test.</li>
<li>testProcess_custom.Rdata: Processed (and integrated if necessary)
Seurat object.</li>
<li>testProcess_gsva_cellname_avg_expression.Rdata: list of cell-type
markers from CelllMarker and Panglao using the GSVA method as well as
the average expression of each gene in each cell-type.</li>
<li>testProcess_cellLabelInfo.Rdata: matrix where each row is a cluster
and each column provides information on the cell-type. Columns provide
info on the cluster from seurat, the cell-type label from CellMarker and
Panglao using the fisher’s exact test and GSVA, and the top 30 markers
per cluser.</li>
</ul>
<div id="interpreting-the-signature-matrix." class="section level4">
<h4>Interpreting the signature matrix.</h4>
<ul>
<li>Each row is a cell-type marker.</li>
<li>Each column is a different cell-type. Assuming that
<code>process_dgTMatrix_lists</code> was used and cell-types were
predicted within scMappR, Each cell-type is labelled as followed.
<ul>
<li>cluster X. “cell-type predicted by Cellmarker” and_or “cell-type
predicted by panglao”.</li>
</ul></li>
</ul>
</div>
<div id="processing-scrna-seq-data-with-multiple-scrna-seq-runs." class="section level3">
<h3>Processing scRNA-seq data with multiple scRNA-seq runs.</h3>
<p>When there are multiple scRNA-seq runs, we use the integration
anchors feature to combine runs. Here, each run is a different element
of the <code>dgTMatrix_list</code> parameter.</p>
<p>Making multiple run scRNA-seq data. As seen in many cases, the first
dataset is identified with “.1” and the second with “.2”</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generating scRNA-seq data with multiple runs.</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sm)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>sm1 <span class="ot">&lt;-</span> sm2 <span class="ot">&lt;-</span> sm</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sm1) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(sm1), <span class="st">&quot;.1&quot;</span>)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sm2) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(sm2),<span class="st">&quot;.2&quot;</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>combined_counts <span class="ot">&lt;-</span> <span class="fu">cbind</span>(sm1,sm2)</span></code></pre></div>
<p>Combining datasets with the integration anchors batch correction</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>toProcess <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) {</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  toProcess[[<span class="fu">paste0</span>(<span class="st">&quot;example&quot;</span>,i)]] <span class="ot">&lt;-</span> combined_counts[,<span class="fu">grep</span>(<span class="fu">paste0</span>(<span class="st">&quot;.&quot;</span>,i), <span class="fu">colnames</span>(combined_counts))]</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>tst1 <span class="ot">&lt;-</span> <span class="fu">process_dgTMatrix_lists</span>(toProcess, <span class="at">name =</span> <span class="st">&quot;testProcess&quot;</span>, <span class="at">species_name =</span> <span class="st">&quot;mouse&quot;</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="at">naming_preference =</span> <span class="st">&quot;eye&quot;</span>, <span class="at">rda_path =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="at">toSave =</span> <span class="cn">TRUE</span>, <span class="at">saveSCObject =</span> <span class="cn">TRUE</span>, <span class="at">path =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
<p>If you do not want to use the integration anchors feature and process
the scRNA-seq data as if it were one run, then keep all of these data in
the same dgCMatrix matrix.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>tst1 <span class="ot">&lt;-</span> <span class="fu">process_dgTMatrix_lists</span>(combined_counts, <span class="at">name =</span> <span class="st">&quot;testProcess&quot;</span>, <span class="at">species_name =</span> <span class="st">&quot;mouse&quot;</span>,</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="at">naming_preference =</span> <span class="st">&quot;eye&quot;</span>, <span class="at">rda_path =</span> <span class="st">&quot;&quot;</span>, </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="at">toSave =</span> <span class="cn">TRUE</span>, <span class="at">saveSCObject =</span> <span class="cn">TRUE</span>, <span class="at">path =</span> <span class="fu">tempdir</span>())</span></code></pre></div>
</div>
<div id="processing-scrna-seq-count-data-from-a-different-species." class="section level3">
<h3>Processing scRNA-seq count data from a different species.</h3>
<p>Please continue to use the <code>process_from_count</code> function.
The only species specific information required in this function is the
mitochondrial gene symbol to regress out mitochondria. In each of your
count matrices, make sure that your michondial genes have a “MT-”
preface. Then select “human” as the species. This can also be done with
an “mt-” preface and selecting mouse.</p>
</div>
<div id="processing-scrna-seq-count-data-when-cell-types-are-already-named." class="section level3">
<h3>Processing scRNA-seq count data when cell-types are already
named.</h3>
<p>It may be common to generate a signature matrix when clusters and
cell-types have already been given for every cell. These examples follow
how to make this signature matrix from:</p>
<ol style="list-style-type: decimal">
<li>A Seurat object with named cell-types</li>
<li>A Count matrix with named cell-types.</li>
</ol>
</div>
<div id="signature-matrix-from-seurat-object-with-named-cell-type" class="section level3">
<h3>Signature matrix from Seurat object with named cell-type</h3>
<p>Generating the Seurat Object for example and making up cell-types.
This example will be used from 1-2.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sm)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>toProcess <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">sm =</span> sm)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>seurat_example <span class="ot">&lt;-</span> <span class="fu">process_from_count</span>(toProcess, <span class="st">&quot;test_vignette&quot;</span>,<span class="at">theSpecies  =</span> <span class="st">&quot;mouse&quot;</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(seurat_example<span class="sc">@</span>active.ident) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Myoblast&quot;</span>, <span class="st">&quot;Neutrophil&quot;</span>, <span class="st">&quot;cardiomyoblast&quot;</span>, <span class="st">&quot;Mesothelial&quot;</span>)</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>A Seurat object with named cell-types. Markers for each cell-type
are stored in the <code>generes</code> object and each signature matrix
is in <code>gene_out</code>.</li>
</ol>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>    generes <span class="ot">&lt;-</span> <span class="fu">seurat_to_generes</span>(<span class="at">pbmc =</span> seurat_example, <span class="at">test =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>    gene_out <span class="ot">&lt;-</span> <span class="fu">generes_to_heatmap</span>(generes, <span class="at">make_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>A count matrix with named cell-types.</li>
</ol>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create the cell-type ids and matrix</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>Cell_type_id <span class="ot">&lt;-</span> seurat_example<span class="sc">@</span>active.ident</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>count_file <span class="ot">&lt;-</span> sm</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>rownames_example <span class="ot">&lt;-</span> <span class="fu">get_gene_symbol</span>(count_file)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(count_file) <span class="ot">&lt;-</span> rownames_example<span class="sc">$</span>rowname</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make seurat object</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>seurat_example <span class="ot">&lt;-</span> <span class="fu">process_from_count</span>(count_file, <span class="st">&quot;test_vignette&quot;</span>,<span class="at">theSpecies  =</span> <span class="st">&quot;mouse&quot;</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Intersect column names (cell-types) with labelled CTs</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>inters <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">colnames</span>(seurat_example), <span class="fu">names</span>(Cell_type_id))</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>seurat_example_inter <span class="ot">&lt;-</span> seurat_example[,inters]</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>Cell_type_id_inter <span class="ot">&lt;-</span> Cell_type_id[inters]</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>seurat_example_inter<span class="sc">@</span>active.ident <span class="ot">&lt;-</span> Cell_type_id_inter</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Making signature matrices</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    generes <span class="ot">&lt;-</span> <span class="fu">seurat_to_generes</span>(<span class="at">pbmc =</span> seurat_example_inter, <span class="at">test =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    gene_out <span class="ot">&lt;-</span> <span class="fu">generes_to_heatmap</span>(generes, <span class="at">make_names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="manually-making-graphics." class="section level2">
<h2>Manually making graphics.</h2>
<p>scMappR generates heatmaps and barplots. The barplots are generated
with <code>plotBP</code> and <code>make_TF_barplot</code>. The plotting
code for <code>plotBP</code> is provided. Inputs are a matrix called
<code>ordered_back_all</code> of -log10(padj) and term names with the
column names log10 and term_name respectively.</p>
<div id="barplots" class="section level3">
<h3>Barplots</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># making an example matrix</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>term_name <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;one&quot;</span>, <span class="st">&quot;two&quot;</span>, <span class="st">&quot;three&quot;</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>log10 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="dv">4</span>, <span class="fl">2.1</span>)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>ordered_back_all <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(term_name,log10))</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a> g <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(ordered_back_all, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x =</span> stats<span class="sc">::</span><span class="fu">reorder</span>(term_name, </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>        log10), <span class="at">y =</span> log10)) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>, </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> <span class="st">&quot;turquoise&quot;</span>) <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">coord_flip</span>() <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;-log10(Padj)&quot;</span>, </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="st">&quot;Gene Ontology&quot;</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> g <span class="sc">+</span> ggplot2<span class="sc">::</span><span class="fu">theme</span>(<span class="at">axis.text.x =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">face =</span> <span class="cn">NULL</span>, </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">35</span>), <span class="at">axis.text.y =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">face =</span> <span class="cn">NULL</span>, </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">angle =</span> <span class="dv">35</span>), <span class="at">axis.title =</span> ggplot2<span class="sc">::</span><span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;black&quot;</span>))</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(y)</span></code></pre></div>
</div>
<div id="heatmaps" class="section level3">
<h3>Heatmaps</h3>
<p>Here, the heatmaps are for plotting cwFold-changes and cell-type
proportions. The same heatmap is used so just an example of one is
given.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generating a heatmap</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Acquiring the gene list</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(POA_example)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>Signature <span class="ot">&lt;-</span> POA_example<span class="sc">$</span>POA_Rank_signature</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>rowname <span class="ot">&lt;-</span> <span class="fu">get_gene_symbol</span>(Signature)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Signature) <span class="ot">&lt;-</span> rowname<span class="sc">$</span>rowname</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(Signature)[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>]</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co">#running tisue_scMappR_custom</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>internal <span class="ot">&lt;-</span> <span class="fu">tissue_scMappR_custom</span>(genes,Signature,<span class="at">output_directory =</span> <span class="st">&quot;scMappR_Test_custom&quot;</span>, <span class="at">toSave =</span> F)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>toPlot <span class="ot">&lt;-</span> internal<span class="sc">$</span>gene_list_heatmap<span class="sc">$</span>geneHeat</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Plotting the heatmap</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>cex <span class="ot">=</span> <span class="fl">0.2</span> <span class="co"># size of genes</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>myheatcol <span class="ot">&lt;-</span> grDevices<span class="sc">::</span><span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;lightblue&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;orange&quot;</span>))(<span class="dv">256</span>)</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>    pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(<span class="fu">as.matrix</span>(toPlot), <span class="at">color =</span> myheatcol, <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, <span class="at">fontsize_row =</span> cex, <span class="at">fontsize_col =</span> <span class="dv">10</span>)</span></code></pre></div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
